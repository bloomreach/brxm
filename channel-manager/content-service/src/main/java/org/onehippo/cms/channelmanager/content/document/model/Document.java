/*
 * Copyright 2016-2019 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.onehippo.cms.channelmanager.content.document.model;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * This bean represents a document, stored in the CMS. It can be serialized into JSON to expose it through a REST API.
 */
@JsonInclude(Include.NON_NULL)
public class Document {
    private String id;                // UUID
    private String variantId;
    private String branchId;
    private String displayName;
    private String urlName;
    private String repositoryPath;
    private DocumentInfo info;        // read-only information about (the current state of) the document
    private Map<String, List<FieldValue>> fields;
    private String state;

    public Document() {
        setInfo(new DocumentInfo());
        setFields(new HashMap<>());
    }

    public String getId() {
        return id;
    }

    public void setId(final String id) {
        this.id = id;
    }

    public String getVariantId() {
        return variantId;
    }

    public void setVariantId(final String variantId) {
        this.variantId = variantId;
    }

    public String getBranchId() {
        return branchId;
    }

    public void setBranchId(final String branchId) {
        this.branchId = branchId;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(final String displayName) {
        this.displayName = displayName;
    }

    public String getUrlName() {
        return urlName;
    }

    public void setUrlName(final String urlName) {
        this.urlName = urlName;
    }

    public String getRepositoryPath() {
        return repositoryPath;
    }

    public void setRepositoryPath(final String repositoryPath) {
        this.repositoryPath = repositoryPath;
    }

    public DocumentInfo getInfo() {
        return info;
    }

    public void setInfo(final DocumentInfo info) {
        this.info = info;
    }

    public Map<String, List<FieldValue>> getFields() {
        return fields;
    }

    public void setFields(final Map<String, List<FieldValue>> fields) {
        this.fields = fields;
    }

    public String getState() {
        return state;
    }

    public void setState(final String state) {
        this.state = state;
    }
}
