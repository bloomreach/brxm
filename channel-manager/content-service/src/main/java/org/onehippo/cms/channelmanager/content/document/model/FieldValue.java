/*
 * Copyright 2016-2018 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.onehippo.cms.channelmanager.content.document.model;

import java.util.List;
import java.util.Map;
import java.util.Objects;
import java.util.Optional;

import org.onehippo.cms.channelmanager.content.documenttype.field.validation.ValidationErrorInfo;

import com.fasterxml.jackson.annotation.JsonInclude;

/**
 * FieldValue encapsulates the value of a field.
 *
 * If the field is primitive, its value is stored in the "value" field. If it is a compound, the child
 * values are stored in the "fields" map. If a field value has validation errors, they can be associated
 * with the field value by storing them inside the errorInfo. The ID field is for future use: when the
 * client would like to reorder fields, each field value must have an ID to recognize a changed value order.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public class FieldValue {
    private String value;                         // Stringified value for primitive fields
    private String id;                            // UUID of the field, if needed
    private Map<String, Object> metadata;         // Optional metadata
    private Map<String, List<FieldValue>> fields; // Subfields of compound
    private String chosenId;                      // ID of a choice in case of a choice field
    private FieldValue chosenValue;               // Singular value of a choice field

    // private String id;    // ID for reordering, unique within a level (list) of field values
    private ValidationErrorInfo errorInfo;

    public FieldValue() { }

    public FieldValue(final String value) {
        this.value = value;
    }

    public FieldValue(final Map<String, List<FieldValue>> fields) {
        this.fields = fields;
    }

    public FieldValue(final String choiceId, final FieldValue choiceValue) {
        this.chosenId = choiceId;
        this.chosenValue = choiceValue;
    }

    public Optional<String> findValue() {
        return Optional.ofNullable(value);
    }

    public boolean hasValue() {
        return value != null;
    }

    // use for serialization and testing only, to avoid NPE.
    public String getValue() {
        return value;
    }

    public void setValue(final String value) {
        this.value = value;
    }

    public String getId() {
        return id;
    }

    public void setId(final String id) {
        this.id = id;
    }

    public Map<String, Object> getMetadata() {
        return metadata;
    }

    public void setMetadata(final Map<String, Object> metadata) {
        this.metadata = metadata;
    }

    public Optional<Map<String, List<FieldValue>>> findFields() {
        return Optional.ofNullable(fields);
    }

    // use for serialization and testing only, to avoid NPE.
    public Map<String, List<FieldValue>> getFields() {
        return fields;
    }

    public void setFields(final Map<String, List<FieldValue>> fields) {
        this.fields = fields;
    }

    public Optional<String> findChosenId() {
        return Optional.ofNullable(chosenId);
    }

    // use for serialization and testing only, to avoid NPE.
    public String getChosenId() {
        return chosenId;
    }

    public void setChosenId(final String chosenId) {
        this.chosenId = chosenId;
    }

    public Optional<FieldValue> findChosenValue() {
        return Optional.ofNullable(chosenValue);
    }

    // use for serialization and testing only, to avoid NPE.
    public FieldValue getChosenValue() {
        return chosenValue;
    }

    public void setChosenValue(final FieldValue chosenValue) {
        this.chosenValue = chosenValue;
    }

    public ValidationErrorInfo getErrorInfo() {
        return errorInfo;
    }

    public void setErrorInfo(final ValidationErrorInfo errorInfo) {
        this.errorInfo = errorInfo;
    }

    public boolean hasErrorInfo() {
        return errorInfo != null;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        final FieldValue that = (FieldValue) o;

        // only include the 'value' fields so the comparison ignores different meta-data of different variants
        return Objects.equals(value, that.value) &&
                Objects.equals(fields, that.fields) &&
                Objects.equals(chosenId, that.chosenId) &&
                Objects.equals(chosenValue, that.chosenValue);
    }

    @Override
    public int hashCode() {
        return Objects.hash(value, fields, chosenId, chosenValue);
    }
}
