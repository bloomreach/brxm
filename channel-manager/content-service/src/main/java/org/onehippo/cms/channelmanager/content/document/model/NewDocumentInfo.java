/*
 * Copyright 2017-2018 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.onehippo.cms.channelmanager.content.document.model;

public class NewDocumentInfo {

    private String name;
    private String slug;
    private String documentTemplateQuery;
    private String documentTypeId;
    private String folderTemplateQuery;
    private String rootPath;
    private String defaultPath;

    public String getName() {
        return name;
    }

    public void setName(final String name) {
        this.name = name;
    }

    public String getSlug() {
        return slug;
    }

    public void setSlug(final String slug) {
        this.slug = slug;
    }

    public String getDocumentTemplateQuery() {
        return documentTemplateQuery;
    }

    public void setDocumentTemplateQuery(final String documentTemplateQuery) {
        this.documentTemplateQuery = documentTemplateQuery;
    }

    public String getDocumentTypeId() {
        return documentTypeId;
    }

    public void setDocumentTypeId(final String documentTypeId) {
        this.documentTypeId = documentTypeId;
    }

    public String getFolderTemplateQuery() {
        return folderTemplateQuery;
    }

    public void setFolderTemplateQuery(final String folderTemplateQuery) {
        this.folderTemplateQuery = folderTemplateQuery;
    }

    public String getRootPath() {
        return rootPath;
    }

    public void setRootPath(final String rootPath) {
        this.rootPath = rootPath;
    }

    public String getDefaultPath() {
        return defaultPath;
    }

    public void setDefaultPath(final String defaultPath) {
        this.defaultPath = defaultPath;
    }
}
