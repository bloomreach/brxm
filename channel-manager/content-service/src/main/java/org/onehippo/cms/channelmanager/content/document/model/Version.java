/*
 * Copyright 2020-2021 Bloomreach
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */
package org.onehippo.cms.channelmanager.content.document.model;

import java.util.Calendar;

import org.onehippo.repository.campaign.Campaign;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class Version {

    private final Calendar timestamp;
    private final String userName;
    private final String jcrUUID;
    private final String branchId;
    private String label;
    private Campaign campaign;

    @JsonCreator
    public Version(
            @JsonProperty("timestamp") Calendar timestamp,
            @JsonProperty("userName") String userName,
            @JsonProperty("jcrUUID") String jcrUUID,
            @JsonProperty("branchId") String branchId,
            @JsonProperty("label") String label,
            @JsonProperty("campaign") Campaign campaign) {
        this.timestamp = timestamp;
        this.userName = userName;
        this.jcrUUID = jcrUUID;
        this.branchId = branchId;
        this.label = label;
        this.campaign = campaign;
    }

    public Calendar getTimestamp() {
        return timestamp;
    }

    public String getUserName() {
        return userName;
    }

    public String getJcrUUID() {
        return jcrUUID;
    }

    public String getBranchId() {
        return branchId;
    }

    public String getLabel() {
        return label;
    }

    public Campaign getCampaign() {
        return campaign;
    }
}
