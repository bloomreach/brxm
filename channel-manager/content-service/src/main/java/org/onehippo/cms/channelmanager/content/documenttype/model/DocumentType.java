/*
 * Copyright 2016-2018 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.onehippo.cms.channelmanager.content.documenttype.model;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

import org.onehippo.cms.channelmanager.content.documenttype.field.type.FieldType;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;


/**
 * This bean represents a document type, known to the CMS.
 * It can be serialized into JSON to expose it through a REST API.
 */
@JsonInclude(Include.NON_DEFAULT)
public class DocumentType {
    private String id; // "namespace:typename"
    private String displayName;
    private boolean readOnlyDueToUnknownValidator;
    private boolean allFieldsIncluded;
    private boolean canCreateAllRequiredFields;
    private final List<FieldType> fields; // ordered list of fields

    @JsonInclude(Include.NON_EMPTY)
    private Set<String> unsupportedFieldTypes = null; // for reporting purposes

    @JsonInclude(Include.NON_EMPTY)
    private Set<String> unsupportedRequiredFieldTypes = null; // for reporting purposes

    public DocumentType() {
        fields = new ArrayList<>();
    }

    public String getId() {
        return id;
    }

    public void setId(final String id) {
        this.id = id;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(final String displayName) {
        this.displayName = displayName;
    }

    public boolean isReadOnlyDueToUnknownValidator() {
        return readOnlyDueToUnknownValidator;
    }

    public void setReadOnlyDueToUnknownValidator(final boolean readOnlyDueToUnknownValidator) {
        this.readOnlyDueToUnknownValidator = readOnlyDueToUnknownValidator;
    }

    public boolean isAllFieldsIncluded() {
        return allFieldsIncluded;
    }

    public void setAllFieldsIncluded(final boolean allFieldsIncluded) {
        this.allFieldsIncluded = allFieldsIncluded;
    }

    public List<FieldType> getFields() {
        return fields;
    }

    public Set<String> getUnsupportedFieldTypes() {
        return unsupportedFieldTypes;
    }

    public void setUnsupportedFieldTypes(final Set<String> unsupportedFieldTypes) {
        this.unsupportedFieldTypes = unsupportedFieldTypes;
    }

    public boolean getCanCreateAllRequiredFields() {
        return canCreateAllRequiredFields;
    }

    public void setCanCreateAllRequiredFields(final boolean canCreateAllRequiredFields) {
        this.canCreateAllRequiredFields = canCreateAllRequiredFields;
    }

    public Set<String> getUnsupportedRequiredFieldTypes() {
        return unsupportedRequiredFieldTypes;
    }

    public void setUnsupportedRequiredFieldTypes(final Set<String> unsupportedRequiredFieldTypes) {
        this.unsupportedRequiredFieldTypes = unsupportedRequiredFieldTypes;
    }
}
