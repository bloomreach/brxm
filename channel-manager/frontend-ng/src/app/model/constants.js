/*
 * Copyright 2020 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// Should match HST's org.hippoecm.hst.core.channelmanager.ChannelManagerConstants.
export const CHANNEL_ID = 'HST-Channel-Id';
export const CONTEXT_PATH = 'HST-Context-Path';
export const DEFAULT_RENDER_VARIANT = 'hippo-default';
export const END_MARKER = 'HST-End';
export const EXPERIENCE_PAGE = 'HST-Experience-Page';
export const EXPERIENCE_PAGE_COMPONENT = 'HST-Experience-Page-Component';
export const EXPERIMENT_ID = 'Targeting-experiment-id';
export const EXPERIMENT_STATE = 'Targeting-experiment-state';
export const HAS_NO_DOM = 'hasNoDom';
export const HAS_PREVIEW_CONFIGURATION = 'HST-Site-HasPreviewConfig';
export const HEAD_ELEMENTS = 'headElements';
export const INHERITED = 'HST-Inherited';
export const LABEL = 'HST-Label';
export const LAST_MODIFIED = 'HST-LastModified';
export const LOCKED_BY = 'HST-LockedBy';
export const LOCKED_BY_CURRENT_USER = 'HST-LockedBy-Current-User';
export const MOUNT_ID = 'HST-Mount-Id';
export const PAGE_ID = 'HST-Page-Id';
export const PATH_INFO = 'HST-Path-Info';
export const REFERENCE_NAMESPACE = 'refNS';
export const RENDER_URL = 'url';
export const RENDER_VARIANT = 'HST-Render-Variant';
export const SHARED = 'HST-Shared';
export const SITEMAPITEM_ID = 'HST-SitemapItem-Id';
export const TYPE = 'HST-Type';
export const TYPE_COMPONENT = 'CONTAINER_ITEM_COMPONENT';
export const TYPE_CONTAINER = 'CONTAINER_COMPONENT';
export const TYPE_EDIT_MENU_LINK = 'EDIT_MENU_LINK';
export const TYPE_MANAGE_CONTENT_LINK = 'MANAGE_CONTENT_LINK';
export const TYPE_PAGE_META = 'PAGE-META-DATA';
export const TYPE_PROCESSED_HEAD_CONTRIBUTIONS = 'HST_PROCESSED_HEAD_CONTRIBUTIONS';
export const TYPE_UNPROCESSED_HEAD_CONTRIBUTIONS = 'HST_UNPROCESSED_HEAD_CONTRIBUTIONS';
export const XTYPE = 'HST-XType';
export const XTYPE_NOMARKUP = 'HST.NoMarkup';
export const XTYPE_SPAN = 'HST.Span';
