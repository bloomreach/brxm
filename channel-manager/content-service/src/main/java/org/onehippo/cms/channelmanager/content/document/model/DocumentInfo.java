/*
 * Copyright 2016-2020 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.onehippo.cms.channelmanager.content.document.model;

import java.util.List;
import java.util.Objects;

import org.onehippo.cms.channelmanager.content.documenttype.model.DocumentType;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * This bean carries information of a document, stored in the CMS.
 * It is part of a document and can be serialized into JSON to expose it through a REST API.
 * Type {@code type} attribute refers to the document's {@link DocumentType} by id.
 */
@JsonInclude(JsonInclude.Include.NON_DEFAULT)
// only return the error count and error messages, never let clients set it
@JsonIgnoreProperties(value = {"errorCount", "errorMessages"}, allowGetters = true)
public class DocumentInfo {

    /**
     * enveloped reference to document type: { id: "namespace:typename" }
     */
    private Type type;

    // whether this document has auto-drafted changes that have not been saved to the preview variant yet
    private boolean dirty;

    // the number of validation errors
    private int errorCount;

    // human-readable messages of document-level validation errors
    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    private List<String> errorMessages;

    // maps to hippostd:publishableSummary (new, live, changed, or unknown)
    private PublicationState publicationState;

    private boolean canPublish;
    private boolean canRequestPublication;
    private String locale;
    private boolean canKeepDraft;
    private boolean retainable;

    public boolean isRetainable() {
        return retainable;
    }

    public Type getType() {
        return type;
    }

    public void setTypeId(final String id) {
        type = new Type(id);
    }

    public void setCanKeepDraft(final boolean saveDraft) {
        this.canKeepDraft = saveDraft;
    }

    public void setRetainable(final boolean retainable) {
        this.retainable = retainable;
    }

    public static class Type {

        private final String id;

        @JsonCreator
        public Type(@JsonProperty("id") String id) {
            this.id = id;
        }

        public String getId() {
            return id;
        }
    }

    public boolean isDirty() {
        return dirty;
    }

    public void setDirty(final boolean dirty) {
        this.dirty = dirty;
    }

    public int getErrorCount() {
        return errorCount;
    }

    public void setErrorCount(final int errorCount) {
        this.errorCount = errorCount;
    }

    public List<String> getErrorMessages() {
        return errorMessages;
    }

    public void setErrorMessages(final List<String> errorMessages) {
        this.errorMessages = errorMessages;
    }

    public PublicationState getPublicationState() {
        return publicationState;
    }

    public void setPublicationState(final PublicationState publicationState) {
        this.publicationState = publicationState;
    }

    public boolean isCanPublish() {
        return canPublish;
    }

    public boolean isCanKeepDraft(){
        return canKeepDraft;
    }

    public boolean isCanRequestPublication() {
        return canRequestPublication;
    }

    public void setCanPublish(final boolean canPublish) {
        this.canPublish = canPublish;
    }

    public void setCanRequestPublication(final boolean canRequestPublication) {
        this.canRequestPublication = canRequestPublication;
    }

    public String getLocale() {
        return locale;
    }

    public void setLocale(final String locale) {
        this.locale = locale;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof DocumentInfo)) {
            return false;
        }
        final DocumentInfo that = (DocumentInfo) o;
        return isDirty() == that.isDirty() &&
                getErrorCount() == that.getErrorCount() &&
                isCanPublish() == that.isCanPublish() &&
                isCanRequestPublication() == that.isCanRequestPublication() &&
                isCanKeepDraft() == that.isCanKeepDraft() &&
                isRetainable() == that.isRetainable() &&
                Objects.equals(getType(), that.getType()) &&
                Objects.equals(getErrorMessages(), that.getErrorMessages()) &&
                getPublicationState() == that.getPublicationState() &&
                Objects.equals(getLocale(), that.getLocale());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getType(), isDirty(), getErrorCount(), getErrorMessages(), getPublicationState(), isCanPublish(), isCanRequestPublication(), getLocale(), isCanKeepDraft(), isRetainable());
    }


    @Override
    public String toString() {
        return "DocumentInfo{" +
                "type=" + type +
                ", dirty=" + dirty +
                ", errorCount=" + errorCount +
                ", errorMessages=" + errorMessages +
                ", publicationState=" + publicationState +
                ", canPublish=" + canPublish +
                ", canRequestPublication=" + canRequestPublication +
                ", locale='" + locale + '\'' +
                ", canKeepDraft=" + canKeepDraft +
                ", retainable=" + retainable +
                '}';
    }
}
