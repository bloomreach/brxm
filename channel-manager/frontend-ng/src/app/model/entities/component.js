/*
 * Copyright 2020 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as HstConstants from '../constants';
import { ComponentEntity } from './component-entity';

export class Component extends ComponentEntity {
  hasExperiment() {
    return !!this.getExperimentId();
  }

  getExperimentId() {
    return this._meta[HstConstants.EXPERIMENT_ID];
  }

  getExperimentStateLabel() {
    if (!this.hasExperiment()) {
      return null;
    }

    return `EXPERIMENT_LABEL_${this._meta[HstConstants.EXPERIMENT_STATE]}`;
  }

  getContainer() {
    return this.container;
  }

  setContainer(container) {
    this.container = container;
  }

  getType() {
    return 'component';
  }

  getRenderVariant() {
    return this._meta[HstConstants.RENDER_VARIANT] || HstConstants.DEFAULT_RENDER_VARIANT;
  }

  getReferenceNamespace() {
    return this._meta[HstConstants.REFERENCE_NAMESPACE];
  }
}
