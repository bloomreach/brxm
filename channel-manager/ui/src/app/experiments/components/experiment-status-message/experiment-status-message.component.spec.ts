/*
 * Copyright 2020-2023 Bloomreach
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Pipe, PipeTransform } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ExperimentVariant } from '../../models/experiment-variant.model';

import { ExperimentStatusMessageComponent } from './experiment-status-message.component';

@Pipe({name: 'translate'})
export class TranslateMockPipe implements PipeTransform {
  transform(value: string, params: { [key: string]: string | number }): string {
    return value + JSON.stringify(params);
  }
}

describe('ExperimentStatusMessageComponent', () => {
  let fixture: ComponentFixture<ExperimentStatusMessageComponent>;
  let component: ExperimentStatusMessageComponent;

  const mockWinnerVariant: ExperimentVariant = {
    variantId: 'variant-1',
    variantName: 'Variant1',
    confidence: 0.7,
    mean: 0.7,
    variance: 0.3,
    visitorSegment: 'segment1',
  };

  beforeEach(() => {
    fixture = TestBed.configureTestingModule({
      declarations: [
        ExperimentStatusMessageComponent,
        TranslateMockPipe,
      ],
    }).createComponent(ExperimentStatusMessageComponent);

    component = fixture.componentInstance;

    component.visits = 100;

    fixture.detectChanges();
  });

  describe('if the winner variant is not present', () => {
    it('should be shown', () => {
      expect(fixture.nativeElement).toMatchSnapshot();
    });
  });

  describe('if the winner variant is present', () => {
    beforeEach(() => {
      component.winnerVariant = mockWinnerVariant;

      fixture.detectChanges();
    });

    it('should be shown', () => {
      expect(fixture.nativeElement).toMatchSnapshot();
    });
  });
});
