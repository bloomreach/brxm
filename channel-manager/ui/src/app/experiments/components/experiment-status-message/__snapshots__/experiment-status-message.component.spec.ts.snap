// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentStatusMessageComponent if the winner variant is not present should be shown 1`] = `
<div
  id="root0"
>
  
  
   EXPERIMENT_WINNER_ABSENTundefined,

  
   EXPERIMENT_AFTER_THIS_MANY_VISITS{"count":100}.

</div>
`;

exports[`ExperimentStatusMessageComponent if the winner variant is present should be shown 1`] = `
<div
  id="root1"
>
  
  
  
   EXPERIMENT_WINNER_PRESENT{"variantName":"Variant1"},

   EXPERIMENT_AFTER_THIS_MANY_VISITS{"count":100}.

</div>
`;
